lxd/instances_post.go

```diff
@@ -711,14 +711,14 @@ func createFromBackup(s *state.State, r *http.Request, projectName string, data
 		// a post hook that can be run once the instance has been created in the database to run any
 		// storage layer finalisations, and a revert hook that can be run if the instance database load
 		// process fails that will remove anything created thus far.
-		postHook, revertHook, err := pool.CreateInstanceFromBackup(*bInfo, backupFile, nil)
+		postHook, revertHook, err := pool.CreateInstanceFromBackup(bInfo, backupFile, nil)
```

Comment:
Oh I see because you dont want to use the uuids in the backup config file as you want to use new uuids on restore, but the storage driver is the one that generates them now.

Reply:
and the backup volume restoration occurs first before opening the backup.yaml file

Reply:
Right what I suggest we do in this particular case of backup restore is to regenerate the uuids in bInfo in createFromBackup before calling pool.CreateInstanceFromBackup, then that will use the new uuids and it also calls backup.UpdateInstanceConfig() so that will get written to the backup.yaml file on disk for  internalImportFromBackup to use so no need to pass it.

Reply:
Along with a nice comment explaining why we are doing this.

Reply:
Thanks for this suggestion! Much cleaner, and no need any more to pass the pointer.

Reply:
This is also removed from the PR.